app/model/build.js

```diff
@@ -1,13 +1,73 @@
+// Imports for packages
 import mongoose from 'mongoose';
+import Promise from 'bluebird';
 
+// Local imports
+import app from '~/';
+
+// Mongoose schema for builds
 const BuildSchema = mongoose.Schema({
   arch:       String,
   target:     String,
   version:    String,
-  started:    Date,
+  started: {
+    type:     Date,
+    default:  Date.now(),
+  },
   finished:   Date,
-  status:     String,
+  status: {
+    type:     String,
+    default: 'QUEUED',
+  },
   log:        String,
 });
 
+// Saves application with updated build information
+// Returns promise of saved application
+BuildSchema.methods.nestSave = function() {
```

Comment:
does mongoose have any functions/abstractions to handle this more automatically?